fil_actor_interface/src/builtin/market/mod.rs

```diff
@@ -163,25 +234,159 @@ pub struct DealProposal {
     pub client_collateral: TokenAmount,
 }
 
-pub enum DealStates<'a, BS> {
-    DealStates(PhantomData<&'a BS>),
+impl From<&fil_actor_market_state::v9::DealProposal> for DealProposal {
+    fn from(deal_proposal: &fil_actor_market_state::v9::DealProposal) -> Self {
+        Self {
+            piece_cid: deal_proposal.piece_cid,
+            piece_size: deal_proposal.piece_size,
+            verified_deal: deal_proposal.verified_deal,
+            client: deal_proposal.client,
+            provider: deal_proposal.provider,
+            label: match &deal_proposal.label {
+                fil_actor_market_state::v9::Label::String(s) => s.clone(),
+                fil_actor_market_state::v9::Label::Bytes(b) => {
+                    String::from_utf8(b.clone()).expect("failed to deserialize utf8 string")
```

Comment:
Are we 100% sure this won't fail? If not, it might make more sense to implement `TryFrom`. Same of the other versions.